azkaban-common/src/main/java/azkaban/executor/container/watch/KubernetesWatch.java

```diff
@@ -75,9 +85,23 @@ public KubernetesWatch(ApiClient apiClient,
     this.podWatchEventListener = podWatchEventListener;
     this.podWatchParams = podWatchParams;
     this.apiClient = apiClient;
-    // no timeout for request completion
+    this.callTimeout =
+        azkProps.getInt(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_CALL_TIMEOUT,
+        DEFAULT_KUBERNETES_WATCH_CALL_TIMEOUT);
+    this.connectTimeout =
+        azkProps.getInt(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_CONNECT_TIMEOUT,
+            DEFAULT_KUBERNETES_WATCH_CONNECT_TIMEOUT);
+    this.readTimeout =
+        azkProps.getInt(ContainerizedDispatchManagerProperties.KUBERNETES_WATCH_READ_TIMEOUT,
+            DEFAULT_KUBERNETES_WATCH_READ_TIMEOUT);
+    // Setting read and call timeouts. So pod watch will be reinitialized.
```

Comment:
Expand this comment elaborate the needs to timeouts a little more. I.e reflect that there are cluster specific setups where we would like to force the reinitialization to handle silent connections drops during the blocking reads.